lib/general.ts

```diff
@@ -26,13 +26,23 @@ export function isValidUrl(url) {
   return true;
 }
 
-/*
+/**
  * Strips off the url parameters from the url path
+ * @param {string} url The url to strip query parameters from
+ * @returns {string} The URL string without the query parameters
  */
-export function urlWithoutParams(url) {
+export function urlWithoutParams(url: string): string {
   return url.split("?")[0];
 }
 
+type JSON =
+  | null
+  | boolean
+  | string
+  | number
+  | Array<JSON>
+  | { [key: string]: JSON };
```

Comment:
I wonder if this should go in globals.d.ts? Not sure if this type is useful outside of this function or not.

Reply:
Yeah not sure either. I was going to keep it this way for now, and as we converted more if I needed `JSON` I would move to a better location